python/llm/src/ipex_llm/transformers/pipeline_parallel.py

```diff
@@ -296,13 +308,16 @@ def pipeline_parallel_generate(self,
                 _past_key_values = past_key_values_placeholder
             else:
                 _past_key_values = outputs.past_key_values
-        elif self.config.model_type in ["baichuan", "chatglm"] and local_rank != 0:
-            # for baichuan2 and chatglm3
-            value_placeholder = torch.empty_like((outputs.past_key_values)[-1][0])
-            past_key_values_placeholder = tuple(
-                (value_placeholder, value_placeholder) for _ in range(layer_start)
-            ) + (outputs.past_key_values)[layer_start:]
-            _past_key_values = past_key_values_placeholder
+        elif self.config.model_type in ["baichuan", "chatglm"] or hasattr(self.config, "visual"):
```

Comment:
or model.config.model_type == "qwen" and hasattr(model.config, "visual")

Reply:
> or model.config.model_type == "qwen" and hasattr(model.config, "visual")

Have updated :)